script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -426,4 +490,51 @@ class PubspecCheckCommand extends PackageLoopingCommand {
     }
     return result ?? Version.none;
   }
+
+  // Validates the dependencies for a package, returning an error string if
+  // there are any that aren't allowed.
+  String? _checkDependencies(Pubspec pubspec) {
+    final Set<String> badDependencies = <String>{};
+    for (final Map<String, Dependency> dependencies
+        in <Map<String, Dependency>>[
+      pubspec.dependencies,
+      pubspec.devDependencies
+    ]) {
+      for (final MapEntry<String, Dependency> entry in dependencies.entries) {
+        final String name = entry.key;
+        if (!_allowDependency(name, entry.value)) {
+          badDependencies.add(name);
+        }
+      }
+    }
+    if (badDependencies.isEmpty) {
+      return null;
+    }
+    return 'The following unexpected non-local dependencies were found:\n'
+        '${badDependencies.map((String name) => '  $name').join('\n')}\n'
+        'Please see https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#Dependencies '
+        'for more information and next steps.';
+  }
+
+  // Checks whether a given dependency is allowed.
+  bool _allowDependency(String name, Dependency dependency) {
```

Comment:
This method name seems a little off from what I would expect. Something like `_isDependancyAllowed` or something similar. 

As is I would expect this to add the dep to an allow list. Up to you if you agree.

Reply:
Good call, changed to `_shouldAllowDependency`